### ISSUE TITLE
TypeError: Invalid Version: v0.1.3a

### ISSUE THREAD

**Issue by lamuertepeluda** [raineorshine/npm-check-updates#758]
- [ x] I have searched for [similar issues](https://github.com/raineorshine/npm-check-updates/issues)
- [ x] I am using the latest version of `npm-check-updates`
- [ x] I am using `node >= 10.17`

---------------------------

ncu 9.2.3
node 14.15.0
npm 6.14.8

[package-files.zip](https://github.com/raineorshine/npm-check-updates/files/5483337/package-files.zip)

## Steps to Reproduce

run against the attached package.json, or 

`npm i git+https://github.com/gcareri/angular-toasty.git#1.0.7`

## Current Behavior

```bash
Checking /home/vito/Development/my-project/package.json
[========------------] 44/112 39%/home/vito/.nvm/versions/node/v14.15.0/lib/node_modules/npm-check-updates/lib/npm-check-updates.js:353
  throw err
  ^

TypeError: Invalid Version: v0.1.3a
    at new SemVer (/home/vito/.nvm/versions/node/v14.15.0/lib/node_modules/npm-check-updates/node_modules/semver/classes/semver.js:41:13)
    at compare (/home/vito/.nvm/versions/node/v14.15.0/lib/node_modules/npm-check-updates/node_modules/semver/functions/compare.js:3:3)
    at Object.gt (/home/vito/.nvm/versions/node/v14.15.0/lib/node_modules/npm-check-updates/node_modules/semver/functions/gt.js:2:29)
    at compareVersions (/home/vito/.nvm/versions/node/v14.15.0/lib/node_modules/npm-check-updates/lib/version-util.js:221:17)
    at Array.sort (<anonymous>)
    at Object.greatest (/home/vito/.nvm/versions/node/v14.15.0/lib/node_modules/npm-check-updates/lib/package-managers/gitTags.js:29:6)
    at async getPackageVersionProtected (/home/vito/.nvm/versions/node/v14.15.0/lib/node_modules/npm-check-updates/lib/versionmanager.js:480:20)
    at async /home/vito/.nvm/versions/node/v14.15.0/lib/node_modules/npm-check-updates/node_modules/p-map/index.js:57:22
```

## Expected Behavior

Prompt the list of packages that can be updated according to the options

Does not happen with ncu 8 (`8.1.11`)

**Comment by lamuertepeluda** [comment]
I forgot

```bash
git ls-remote --tags  https://github.com/gcareri/angular-toasty.git
7c1617ccc23b54aeeb63e6a594a65a79ddc885d2        refs/tags/0.1.2
d40b3f2b1f7260185c98ffacbd0828eedd41a142        refs/tags/1.0.5
cd040c99002b8650e95872c72ce661f527b2f264        refs/tags/1.0.6
5bd70fdcbf442fb5c1f34dac449dc789c8474760        refs/tags/1.0.7
6a6b137902a941fc5c680f7b90a1ed233b7d9db6        refs/tags/v0.1.3
3b04fa253d5a0d7e165154ab2429530391e1f1f4        refs/tags/v0.1.3a
5a42921e5e626d3fb49429dbce0a2b9e235824ed        refs/tags/v0.1.4
f3830e7bb7dc2f4ea2062e6175f658fb39d06709        refs/tags/v0.1.5
6f9d4cbe2fdc52d8343a29ad5d361f683eeda16e        refs/tags/v0.1.6
9f40bfb0d13cfc55a94cabcc1b5c7301f055d7d8        refs/tags/v0.1.7
e86917d3e101c9b84ebe4c94f8aef1c4cb06176c        refs/tags/v0.1.8
46a9dd345791bded859a187dfab35c14ab8dd84c        refs/tags/v1.0.0
3bd8ebb988a8c8b3544ba6150fcebdcd04a42aa5        refs/tags/v1.0.1
b7553f61648c9de80f7f72ad27c2bef82f56a75e        refs/tags/v1.0.2
8cee4fcdf2be350259831053b72458c05dd67ce5        refs/tags/v1.0.3
3dab633f65541b2ba8c113d28eca743016b08c30        refs/tags/v1.0.4
```

**Comment by lamuertepeluda** [comment]
so yeah, definitely bad luck & bad semantic versioning in this package 😅 

I suggest to handle the exception by raising a warning (or lower level debug message) and avoid the complete crash of ncu

**Comment by raineorshine** [comment]
Thanks for the info!

Agreed. Since there is no presumption that git tags are valid semver, it should fail gracefully. I'll look into it.

**Comment by raineorshine** [comment]
Fixed and published in `v9.2.4`
